server/form-pages/apply/health-needs/health-needs/learningDifficultiesDetails.test.ts

```diff
@@ -0,0 +1,135 @@
+      })
+
+      it('includes a validation error for _receivesTreatment_', () => {
+        expect(page.errors()).toHaveProperty('receivesTreatment', 'Select yes if they receive any treatment')
+      })
+
+      it('includes a validation error for _isVulnerable_', () => {
+        expect(page.errors()).toHaveProperty('isVulnerable', 'Select yes if they are vulnerable')
+      })
+    })
+
+    describe('when _needsSupport_ is YES', () => {
+      const page = new LearningDifficultiesDetails({ needsSupport: 'yes' }, application)
+
+      describe('and _interactionDetail_ is UNANSWERED', () => {
+        it('includes a validation error for _supportDetail_', () => {
+          expect(page.errors()).toHaveProperty('supportDetail', 'Enter the type of support needed')
+        })
+      })
+    })
+
+    describe('when _receivesTreatment_ is YES', () => {
+      const page = new LearningDifficultiesDetails({ receivesTreatment: 'yes' }, application)
+
+      describe('and _treatmentDetail_ is UNANSWERED', () => {
+        it('includes a validation error for _treatmentDetail_', () => {
+          expect(page.errors()).toHaveProperty('treatmentDetail', 'Enter details about their treatment')
+        })
+      })
+    })
+
+    describe('when _isVulnerable_ is YES', () => {
+      const page = new LearningDifficultiesDetails({ isVulnerable: 'yes' }, application)
+
+      describe('and _vulnerabilityDetail_ is UNANSWERED', () => {
+        it('includes a validation error for _vulnerabilityDetail_', () => {
+          expect(page.errors()).toHaveProperty('vulnerabilityDetail', 'Enter how they are vulnerable')
+        })
+      })
+    })
+  })
+
+  describe('onSave', () => {
+    it('removes support data when the question is set to "no"', () => {
+      const body: Partial<LearningDifficultiesDetailsBody> = {
+        needsSupport: 'no',
+        supportDetail: 'support detail',
+      }
+
+      const page = new LearningDifficultiesDetails(body, application)
+
+      page.onSave()
+
+      expect(page.body).toEqual({
+        needsSupport: 'no',
+      })
+    })
+
+    it('removes treatment data when the question is set to "no"', () => {
+      const body: Partial<LearningDifficultiesDetailsBody> = {
+        receivesTreatment: 'no',
+        treatmentDetail: 'treatment detail',
+      }
+
+      const page = new LearningDifficultiesDetails(body, application)
+
+      page.onSave()
+
+      expect(page.body).toEqual({
+        receivesTreatment: 'no',
+      })
+    })
+
+    it('removes vulnerability data when the question is set to "no"', () => {
+      const body: Partial<LearningDifficultiesDetailsBody> = {
+        isVulnerable: 'no',
+        vulnerabilityDetail: 'Vulnerability detail',
+      }
+
+      const page = new LearningDifficultiesDetails(body, application)
+
+      page.onSave()
+
+      expect(page.body).toEqual({
+        isVulnerable: 'no',
+      })
+    })
```

Comment:
Quite a lot of code duplication here.  I'm, not sure there's a huge benefit in testing each one of these separately so they could be combined. Either that or again use am `it.each()` block.